clio-server/src/main/resources/swagger/api-docs.yaml

```diff
@@ -0,0 +1,383 @@
+swagger: "2.0"
+
+info:
+  description: "This is the API specification for Clio."
+  version: "1.0.0"
+  title: "Clio"
+  contact:
+    email: "greenteam@broadinstitute.org"
+  license:
+    name: "BSD 3-clause"
+    url: "https://opensource.org/licenses/BSD-3-Clause"
+
+host: "localhost:8080"
+basePath: "/api/v1"
+tags:
+  - name: "readgroup"
+    description: "Read group endpoints"
+
+schemes:
+  - "http"
+
+security:
+  - googleoauth:
+      - "openid"
+      - "email"
+      - "profile"
+
+paths:
+  /readgroup/metadata:
+    post:
+      tags:
+         - "readgroup"
+      summary: "Posts read group metadata."
+      operationId: "readGroupMetadata"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      responses:
+        '200':
+          description: "Successful Request"
+  /readgroup/schema:
+    get:
+      tags:
+        - "readgroup"
+      summary: "Gets the json schema for read groups"
+      operationId: "readGroupSchema"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Read group metadata to store"
+          required: true
+          schema:
+            $ref: '#/definitions/TransferReadGroupV1Metadata'
+      responses:
+        '200':
+          description: "Successful Request"
+      security:
+        - googleoauth:
+          - "openid"
+  /readgroup/query:
+    post:
+      tags:
+        - "readgroup"
+      summary: "Query for a read group"
+      operationId: "readGroupQuery"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Read group query to search for"
+          required: true
+          schema:
+            $ref: '#/definitions/TransferReadGroupV1QueryInput'
+      responses:
+        '200':
+          description: "Successful Request"
+          schema:
+            $ref: '#/definitions/TransferReadGroupV1QueryResponse'
+        '405':
+          description: "Invalid input"
+      security:
+        - googleoauth:
+            - "openid"
+
+securityDefinitions:
+  googleoauth:
+    type: "oauth2"
+    authorizationUrl: 'https://accounts.google.com/o/oauth2/auth'
+    flow: "implicit"
+    scopes:
+      openid: "open id authorization"
+      email: "email authorization"
+      profile: "profile authorization"
+
+definitions:
+  TransferReadGroupV1Key:
```

Comment:
Not sure about this one I haven't touched the models yet. It did seem odd that we have a key, metadata and another that is the key + the metadata.